code/lib/core-common/src/utils/validate-config.ts

```diff
@@ -1,19 +1,48 @@
+import { join } from 'path';
 import { dedent } from 'ts-dedent';
+import { frameworkPackages } from './get-storybook-info';
 
 const renderers = ['html', 'preact', 'react', 'server', 'svelte', 'vue', 'vue3', 'web-components'];
 
 const rendererNames = [...renderers, ...renderers.map((renderer) => `@storybook/${renderer}`)];
 
-// Checks that the framework name is not a renderer
-export function validateFrameworkName(frameworkName: string) {
+export function validateFrameworkName(frameworkName: string | undefined) {
+  const automigrateMessage = `Please run 'npx storybook@next automigrate' to automatically fix your config.
+
+  See the migration guide for more information: 
+  https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#new-framework-api\n`;
+
+  // Throw error if there is no framework field
+  // TODO: maybe this error should not be thrown if we allow empty Storybooks that only use "refs" for composition
```

Comment:
AFAIK this is not something we allow/support.
The UI will show you a "empty storybook" warning if you'd do that.